assets/js/share.js

```diff
@@ -151,10 +170,24 @@ function addExternalShareListeners(externalService) {
 function scrollToSectionInView() {
   const url = new URL(window.location.href);
   const sectionId = url.hash || url.pathname.replace("/", "#");
-  if (sectionId) {
+  const isMobile = window.matchMedia("(max-width: 600px)").matches;
+
+  if (sectionId.length > 1) {
     const sections = document.querySelectorAll("details");
     const clickedSection = document.querySelector(sectionId);
 
+    if (!isMobile) {
+      const oldOpenSection = document.querySelector("details.desktop-open");
```

Comment:
lines 180-188 and 38-45 are alomst identical. can we share the code between them?